src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientConfiguration.java

```diff
@@ -89,4 +89,14 @@
         description = "HTTP Headers which shall be sent with each request. Might be used for authentication. The format of each header is name:value",
         type = AttributeType.STRING)
     String[] httpHeaders();
+
```

Comment:
Would it be possible to make this more user friendly, like having distinct fields for each cache configuration parameter?

I wonder if there is a possibility for clashing between different cache configurations when in the same AEM instance multiple components need to configure caches via content packages.

Reply:
Unfortunately it's not possible to make this more user-friendly. I borrowed the configuration style with `:` separator from many other AEM OSGi services that also do it like that.

I'm not sure what you mean by "clashes", but of course there can only be one configuration and this would not be done by components, this is a global configuration for a site.

Reply:
It's actually possible for the "Apache Sling Logging Logger Configuration" and others.

> this is a global configuration for a site

That will break modularity and introduce a maintenance issue where you need to track separately the cache configuration requirements of each module using this feature to update the global configuration.
Also note the conceptual dependency cycle how graphql-client depends on a module which is using graphql-client.

Reply:
> Apache Sling Logging Logger Configuration
This is a factory config, it's a different type of configuration. Here we have one single client config, with one "multi-field" having multiple parameters. What I did is similar to services like "Apache Sling Scripting HTL JavaScript Use-API Factories Configuration" or "Apache Sling Scripting HTL JavaScript Use-API Factories Configuration".